src/main/java/de/codecentric/centerdevice/util/StageUtils.java

```diff
@@ -46,12 +56,49 @@ public static void focusNextStage() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	public static ObservableList<Stage> getStages() {
-		return StageHelper.getStages();
+		if (stages == null) {
+			// Java 9
+			try {
+				windows = (ObservableList<Window>)Window.class.getMethod("getWindows").invoke(null);
+				stages = FXCollections.observableArrayList();
+				windows.addListener(new ListChangeListener<Window>() {
+					@Override
+					public void onChanged(javafx.collections.ListChangeListener.Change<? extends Window> c) {
+						updateStages();
```

Comment:
Where do you see a memory leak here? "windows" is just a read-only reference to the global list of windows and in the updateStages method I copy all current windows from this list into a temporary stage list and finally "set" this as the new content of "stages". So any closed window will not be in the "windows" list and thus will finally also be removed from the "stages". So there is no reference to closed windows anymore, or am I missing something here?